extensions/jib/deployment/src/main/java/io/quarkus/jib/deployment/JibConfig.java

```diff
@@ -0,0 +1,38 @@
+package io.quarkus.jib.deployment;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(phase = ConfigPhase.BUILD_TIME)
+public class JibConfig {
+
+    /**
+     * If enabled, a container image will be created using jib
+     */
+    @ConfigItem(defaultValue = "true")
+    public boolean enabled;
```

Comment:
We may also need a flag that will specify if we want to perform the build via docker daemon or not.

Jib can perform dockerless builds, but I find it failing more often than I'd like  (due to timeouts, dns etc).
I would like to be able to choose between dockerless and docker daemon builds.

Reply:
That is something I was looking for feedback on. Basically I don't really understand how I can use a container when a dockerless build is being used.

Reply:
The image is pushed to a registry and then your actual container can use it from there.

Reply:
So then we need some configuration that will allow to push to a registry. From a technical point of view this should be easy. But I am still struggling to form a big picture in my mind of how this and the Kubernetes stuff can form a complete, effortless and intuitive user experience

Reply:
@geoand that configuration comes from the environment i reckon - like docker / kubectl gets it from env variables/context setup.